products/BellHybrid/services/audio/ServiceAudio.cpp

```diff
@@ -197,18 +196,23 @@ namespace service
         return std::make_unique<AudioStartPlaybackResponse>(retCode, retToken);
     }
 
-    auto Audio::handleStop([[maybe_unused]] const std::vector<audio::PlaybackType> &stopTypes,
-                           const audio::Token &token) -> std::unique_ptr<AudioResponseMessage>
+    auto Audio::handleStop(const std::vector<audio::PlaybackType> &stopTypes, const audio::Token &token)
+        -> std::unique_ptr<AudioResponseMessage>
     {
         std::vector<std::pair<audio::Token, audio::RetCode>> retCodes;
 
+        // if stopType is not provided then stop all inputs, otherwise stop specific ones from stopType
         for (auto &input : audioMux.GetAllInputs()) {
-            auto t = input.token;
-            if (token.IsValid() && t == token) {
-                retCodes.emplace_back(t, stopInput(&input));
-            }
-            if (token.IsUninitialized()) {
-                retCodes.emplace_back(t, stopInput(&input));
+            const auto &currentOperation = input.audio->GetCurrentOperation();
```

Comment:
I don't really understand why this change is needed, what case requires stopping of the specified input?

Reply:
I get it, thanks for the explanation!